src/System.IO.FileSystem.Watcher.Polling/System/IO/PollingFileSystemWatcher.cs

```diff
@@ -131,12 +133,33 @@ internal void UpdateState(string directory, ref FileChangeList changes, ref File
         /// </summary>
         public void Dispose()
         {
-            _timer.Dispose();
+            Dispose(true);
+            GC.SuppressFinalize(this);
         }
 
         public void Dispose(WaitHandle notifyObject)
         {
-            _timer.Dispose(notifyObject);
+            Dispose(true, notifyObject);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing, WaitHandle notifyObject = null)
+        {
+            if (disposed) return;
```

Comment:
I was actually thinking of checking in the callback and not calling `.Change()` again if disposed.

Reply:
Good idea!